pkg/subctl/cmd/join.go

```diff
@@ -262,10 +262,11 @@ func joinSubmarinerCluster(config *rest.Config, subctlData *datafile.SubctlData)
 
 	if subctlData.GlobalnetCidrRange == "" {
 		// Globalnet not enabled
-		err = checkOverlappingServiceCidr(globalNetworks)
+		err = checkOverlappingCidr(globalNetworks, "service")
```

Comment:
passing a string as a parameter to specify what to check is error prone, the developer could
pass a wrong string and compiler won't detect anything.

At least I'd use consts, 
```go
type OverlapCheck int

const (
             ClusterOverlap OverlapCheck = iota
             ServiceOverlap
             GlobalnetOverlap
)
```